File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        
        Arrays.sort(citations);
        
        for (int i=0;i<citations[citations.length - 1];i++) {
            int low = 0;
            int high = 0;
            int h = i;
            
            for (int j=0;j<citations.length;j++) {
                if (citations[j] >= h) high++;
                
            }
            low = citations.length - high;
            if (high >= h && low <= h) {
                answer = Math.max(answer, h);
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <bits/stdc++.h>

using namespace std;

int l, r, mid;

bool check(int c, vector<int> citations) {
    int ok = 0;
    int no = 0;
    
    for (int i : citations) {
        if (i >= c) ok++;
    }
    no = citations.size() - ok;
    if (ok >= c && no <= c) return true;
    return false;
}

int solution(vector<int> citations) {
    int answer = 0;
    
    sort(citations.begin(), citations.end());
    l = 0;
    r = citations[citations.size() - 1];
    
    while (l <= r) {
        mid = (l + r) / 2;
        
        if (check(mid, citations)) {
            answer = max(answer, mid);
            l = mid + 1;
        } else {
            r = mid - 1;
        }
    }
    
    
    return answer;
}